glotaran/io/register.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import os
+
+import xarray as xr
+
+from glotaran.model import Model
+from glotaran.parameter import ParameterGroup
+from glotaran.project import Result
+from glotaran.project import SavingOptions
+from glotaran.project import Scheme
+
+from .io import Io
+
+_io_register = {}
+
+
+def register_io(fmt: str | list[str], io: Io):
+    fmt = fmt if isinstance(fmt, list) else [fmt]
+    for fmt_name in fmt:
+        _io_register[fmt_name] = io()
+
+
+def known_fmt(fmt: str) -> bool:
+    return fmt in _io_register
+
+
+def get_io(fmt: str):
```

Comment:
This should return a generic type with lower bound `Io`

Reply:
yes